packages/create-strapi-starter/utils/build-starter.js

```diff
@@ -0,0 +1,182 @@
+
+const { runInstall, runApp, initGit, createInitialGitCommit } = require('./child-process');
+
+/**
+ * @param  {string} starterUrl Github url to starter project
+ */
+async function getRepoInfo(starterUrl) {
+  const repoInfo = GitUrlParse(starterUrl);
+  const { name, full_name, ref } = repoInfo;
+
+  const response = await fetch(`https://api.github.com/repos/${full_name}`);
+
+  if (!response.ok) {
+    throw Error(`Could not fetch the default branch`);
+  }
+
+  const { default_branch } = await response.json();
+
+  return {
+    name,
+    full_name,
+    ref,
+    default_branch,
+  };
+}
+
+/**
+ * @param  {string} starterUrl Github url for strapi starter
+ * @param  {string} tmpDir Path to temporary directory
+ */
+async function downloadGithubRepo(starterUrl, tmpDir) {
+  const { name, full_name, ref, default_branch } = await getRepoInfo(starterUrl);
+
+  const branch = ref ? ref : default_branch;
+
+  // Download from GitHub
+  const codeload = `https://codeload.github.com/${full_name}/tar.gz/${branch}`;
+  const response = await fetch(codeload);
+  if (!response.ok) {
+    throw Error(`Could not download the ${chalk.green(`${name}`)} repository`);
+  }
+
+  await new Promise(resolve => {
+    response.body.pipe(tar.extract({ strip: 1, cwd: tmpDir })).on('close', resolve);
+  });
+}
+
+/**
+ * @param  {string} filePath Path to starter.json file
+ */
+function readStarterJson(filePath) {
+  const data = fse.readFileSync(filePath);
+  return JSON.parse(data);
+}
+
+/**
+ * @param  {string} rootPath Path to the project directory
+ * @param  {string} projectName Name of the project
+ */
+function initPackageJson(rootPath, projectName) {
+  const packageManager = hasYarn ? 'yarn' : 'npm run';
+
+  fse.writeJson(join(rootPath, 'package.json'), {
+    name: projectName,
+    private: true,
+    version: '0.0.0',
+    scripts: {
+      'dev:backend': `cd backend && ${packageManager} develop`,
+      'dev:frontend': `wait-on http://localhost:1337/admin && cd frontend && ${packageManager} develop --open`,
+      develop: `concurrently "${packageManager} dev:backend" "${packageManager} dev:frontend"`,
+    },
+    dependencies: {
+      concurrently: '^6.0.0',
+      'wait-on': '^5.2.1',
+    },
+  });
+}
+/**
+ * @param  {string} path The directory path for install
+ */
+async function installWithLogs(path) {
+  const installPrefix = chalk.yellow('Installing dependencies:');
+  const loader = ora(installPrefix).start();
+  const logInstall = (chunk = '') => {
+    loader.text = `${installPrefix} ${chunk
+      .toString()
+      .split('\n')
+      .join(' ')}`;
+  };
+
+  const runner = runInstall(path);
+  runner.stdout.on('data', logInstall);
+  runner.stderr.on('data', logInstall);
+
+  await runner;
+
+  loader.stop();
+  console.log(`Dependencies installed ${chalk.green('successfully')}.`);
+}
+
+module.exports = async function buildStarter(projectName, program) {
+  const starterUrl = program.args[1];
```

Comment:
you can get this in a cleaner way like this

```js
const commander = require('commander');
const packageJson = require('./package.json');

const program = new commander.Command(packageJson.name);

let projectName;

program
  .version(packageJson.version)
  .arguments('<directory> <starterurl>') // define arguments once
  .option(
    '--use-npm',
    'Force usage of npm instead of yarn to create the project'
  )
  .option('--debug', 'Display database connection error')
  .option('--quickstart', 'Quickstart app creation')
  .option('--dbclient <dbclient>', 'Database client')
  .option('--dbhost <dbhost>', 'Database host')
  .option('--dbsrv <dbsrv>', 'Database srv')
  .option('--dbport <dbport>', 'Database port')
  .option('--dbname <dbname>', 'Database name')
  .option('--dbusername <dbusername>', 'Database username')
  .option('--dbpassword <dbpassword>', 'Database password')
  .option('--dbssl <dbssl>', 'Database SSL')
  .option('--dbauth <dbauth>', 'Authentication Database')
  .option('--dbfile <dbfile>', 'Database file path for sqlite')
  .option('--dbforce', 'Overwrite database content if any')
  .description('create a new application')
  .action((directory, starterUrl, program) => { // everything here 
    console.log('building ', directory, starterUrl, program);
    buildproject({ projectName: directory, starterUrl }, program)
  })
  .parse(process.argv);

```